connect/mirror/src/test/java/org/apache/kafka/connect/mirror/MirrorSourceConnectorTaskConfigTest.java

```diff
@@ -0,0 +1,69 @@
+package org.apache.kafka.connect.mirror;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.apache.kafka.common.TopicPartition;
+import org.junit.Test;
+
+import static org.apache.kafka.connect.mirror.MirrorConnectorConfig.TASK_TOPIC_PARTITIONS;
+import static org.junit.Assert.assertEquals;
+
+public class MirrorSourceConnectorTaskConfigTest {
+
+    @Test
+    public void testMirrorSourceConnectorTaskConfig() {
+        MirrorSourceConnector connector = new MirrorSourceConnector();
+
+        List<TopicPartition> knownTopicPartitions = new ArrayList<>();
+
+        // topic `t0` has 8 partitions
+        knownTopicPartitions.add(new TopicPartition("t0", 0));
+        knownTopicPartitions.add(new TopicPartition("t0", 1));
+        knownTopicPartitions.add(new TopicPartition("t0", 2));
+        knownTopicPartitions.add(new TopicPartition("t0", 3));
+        knownTopicPartitions.add(new TopicPartition("t0", 4));
+        knownTopicPartitions.add(new TopicPartition("t0", 5));
+        knownTopicPartitions.add(new TopicPartition("t0", 6));
+        knownTopicPartitions.add(new TopicPartition("t0", 7));
+
+        // topic `t1` has 2 partitions
+        knownTopicPartitions.add(new TopicPartition("t1", 0));
+        knownTopicPartitions.add(new TopicPartition("t1", 1));
+
+        // topic `t2` has 2 partitions
+        knownTopicPartitions.add(new TopicPartition("t2", 0));
+        knownTopicPartitions.add(new TopicPartition("t2", 1));
+
+        // MirrorConnectorConfig example for test
+        Map<String, String> props = new HashMap<>();
+        props.put("enabled", "true");
+        props.put("name", "ConnectorName");
+        props.put("connector.class", "ConnectorClass");
+        props.put("source.cluster.alias", "source1");
+        props.put("target.cluster.alias", "target2");
+        MirrorConnectorConfig config = new MirrorConnectorConfig(props);
+
+        // config MirrorSourceConnector as minimum to run taskConfig()
```

Comment:
instead of these setters, you could add a new constructor to support this test

Reply:
sounds great